File: server/repository/event.go
package repository

type UmiEventLevel string

const (
	UmiEventLevelInfo     UmiEventLevel = "info"
	UmiEventLevelWarning  UmiEventLevel = "warning"
	UmiEventLevelError    UmiEventLevel = "error"
	UmiEventLevelCritical UmiEventLevel = "critical"
)

type UmiEvent struct {
	Id        string            `json:"id"`
	ServiceId string            `json:"service_id"`
	ChannelId string            `json:"channel_id"`
	Title     string            `json:"title"`
	Body      *string           `json:"body"`
	Level     *UmiEventLevel    `json:"level" validate:"omitempty,oneof=info warning error critical"`
	ActionURL *string           `json:"action_url"`
	IconURL   *string           `json:"icon_url"`
	Metadata  map[string]string `json:"metadata"`
	CreatedAt int64             `json:"created_at"`
}

File: web/src/types/event.ts
export type UmiEventLevel = "info" | "warning" | "error" | "critical";

export interface UmiEvent {
    id: string;
    title: string;
    body: string | null;
    level: UmiEventLevel | null;
    action_url: string | null;
    icon_url: string | null;
    tags: string[];
    service: string;
    metadata: Record<string, string> | null;
    created_at: string; // ISO 8601
}

export interface UmiEventPayload {
    title: string;
    body?: string;
    level?: UmiEventLevel;
    action_url?: string;
    icon_url?: string;
    tags: string[];
    metadata?: Record<string, string>;
}
